### PR TITLE
feat: setup replica with RDS on Cloud

### PR DESCRIPTION
**Pull request by TianyuZhang1214**
# What's New

## Documentation

1. **New Guide: [replica-setup-rds.md](docs/tutorial/replica-setup-rds.md)**  
   A new user guide has been added, detailing how to synchronize data from RDS on various cloud platforms. Our tests confirm support for the following:  
   - AWS MySQL
   - AWS Aurora for MySQL
   - Aliyun RDS
   - Azure Database for MySQL
   - ApeCloud WeSQL
   - MySQL Community Edition

## Enhancements

1. **Snapshot Stage Behavior**  
   The snapshot stage is now skipped if the source MySQL server is empty (only contains system databases). This change prevents errors caused by `mysqldump` when no user data is present.
   
2. **New Tool: `checker.sh`**  
   A new script, `checker.sh`, has been introduced to verify that the MySQL configuration is correct and that the user has the required privileges for replication setup.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -110,15 +110,16 @@ docker run \
...
+  --env=SETUP_MODE=COMBINED \
   --env=MYSQL_HOST=<mysql_host> \
   --env=MYSQL_PORT=<mysql_port> \
   --env=MYSQL_USER=<mysql_user> \
   --env=MYSQL_PASSWORD=<mysql_password> \
   --detach=true \
   apecloud/myduckserver:latest
 ```
-
+## Try on Cloud Vendors
```
Comment (fanyang01) [inline README.md:121]: "Try on Cloud Vendors" -> "Connect to Cloud MySQL"

File: docs/tutorial/replica-setup-rds.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
++--------------------------+-------+
+| binlog_format            | ROW   |
+| enforce_gtid_consistency | ON    |
+| gtid_mode                | ON    |
+| log_bin                  | ON    |
++--------------------------+-------+
+```
+
+These parameters are usually enabled by default or easily configurable with most cloud providers like Azure, Aliyun, ApeCloud, and MySQL Community Edition. However, for **AWS RDS** or **AWS Aurora for MySQL**, you may need to follow the specific instructions to enable them. Refer to the official documentation:
+- [AWS MySQL](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_LogAccess.MySQL.BinaryFormat.html)
```
Comment (fanyang01) [inline docs/tutorial/replica-setup-rds.md:27]: Amazon RDS for MySQL

File: docs/tutorial/replica-setup-rds.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
+# MyDuckServer Replica Setup Preparation
```
Comment (fanyang01) [inline docs/tutorial/replica-setup-rds.md:1]: ~~Preparation~~

**Review by fanyang01 — COMMENTED** [review]
LGTM! This PR make MyDuckServer look much more production-ready and useful.

File: devtools/replica-setup/start_snapshot.sh [hunk 4]
```diff
@@ -52,7 +52,7 @@ echo "Thread count set to: $THREAD_COUNT"
 # 3rd step: Execute mysqlsh command
 echo "Starting snapshot copy with mysqlsh..."
 # Run mysqlsh command and capture the output
-output=$(mysqlsh -h${MYSQL_HOST} -P${MYSQL_PORT} -u${MYSQL_USER} -p${MYSQL_PASSWORD} -- util copy-instance 'mysql://admin:admin@127.0.0.1:3306' --users false --ignore-existing-objects true --handle-grant-errors ignore --threads $THREAD_COUNT --bytesPerChunk 256M)
+output=$(mysqlsh -h${MYSQL_HOST} -P${MYSQL_PORT} -u${MYSQL_USER} -p${MYSQL_PASSWORD} -- util copy-instance 'mysql://admin:admin@127.0.0.1:3306' --users false --consistent false --ignore-existing-objects true --handle-grant-errors ignore --threads $THREAD_COUNT --bytesPerChunk 256M)
```
[the remark this answers is not in the document]
Comment (TianyuZhang1214) [reply devtools/replica-setup/start_snapshot.sh:55]: If `--consistent` is enabled, the `LOCK TABLES` privilege is required. However, cloud vendors typically do not grant users sufficient privileges to lock tables, so this feature is disabled by default.

File: devtools/replica-setup/start_delta.sh [hunk 5]
```diff
@@ -1,5 +1,9 @@
 #!/bin/bash
 
+if [[ $SOURCE_IS_EMPTY -eq 0 ]]; then
+  EXECUTED_GTID_SET=$(mysqlsh --host="$MYSQL_HOST" --user="$MYSQL_USER" --password="$MYSQL_PASSWORD" --sql -e "SHOW MASTER STATUS\G" | grep -i "Executed_Gtid_Set" | awk -F': ' '{print $2}')
```
[the remark this answers is not in the document]
Comment (TianyuZhang1214) [reply devtools/replica-setup/start_delta.sh]: fixed

File: docs/tutorial/replica-setup-rds.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  --network=host \
+  --privileged \
+  --workdir=/home/admin \
+  --env=SETUP_MODE=COMBINED \
+  --env=MYSQL_HOST=<mysql_host> \
+  --env=MYSQL_PORT=<mysql_port> \
+  --env=MYSQL_USER=<mysql_user> \
+  --env=MYSQL_PASSWORD=<mysql_password> \
+  --detach=true \
+  tyzhang1214/myduckserver:latest
```
[the remark this answers is not in the document]
Comment (TianyuZhang1214) [reply docs/tutorial/replica-setup-rds.md]: fixed

File: docs/tutorial/replica-setup-rds.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+```
+
+## Step 4: Check Replica Status
+
+To verify the replica setup, connect to MyDuckServer:
+
+```bash
+mysql -h127.0.0.1 -P3306 -uroot -A
```
[the remark this answers is not in the document]
Comment (TianyuZhang1214) [reply docs/tutorial/replica-setup-rds.md]: fixed

File: docs/tutorial/replica-setup-rds.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
+# MyDuckServer Replica Setup Preparation
+
+MyDuckServer supports setting up replicas from a variety of MySQL-based databases, including AWS MySQL, AWS Aurora for MySQL, Aliyun RDS, Azure Database for MySQL, ApeCloud WeSQL, and MySQL Community Edition.
```
[the remark this answers is not in the document]
Comment (TianyuZhang1214) [reply docs/tutorial/replica-setup-rds.md]: fixed

File: README.md (hunk 1 above)
Comment (TianyuZhang1214) [reply README.md]: fixed

File: docs/tutorial/replica-setup-rds.md (hunk 2 above)
Comment (TianyuZhang1214) [reply docs/tutorial/replica-setup-rds.md]: fixed

File: docs/tutorial/replica-setup-rds.md (hunk 3 above)
Comment (TianyuZhang1214) [reply docs/tutorial/replica-setup-rds.md]: fixed

File: devtools/replica-setup/start_snapshot.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (fanyang01) [reply devtools/replica-setup/start_snapshot.sh:55]: I get it.

**Pull request merged**